### DISCUSSION TITLE
GAL22V10 Output Feedback

### DISCUSSION THREAD

**Discussion by jimktrains** [hneemann/Digital#845] [Q&A] [answered]
I'm brand new to working with CPLDs and Digital (and music synthesizers, so all the cards are stacked against me), my apologies if I've missed this somewhere.  I'm trying to build a Top Octave Synthesizer (well, more of a 1-note synthesizer given the I/O constraints), and I have a counter as the heart of the circuit. When I analyze, there are inputs of "Counter_1^n" and outputs of "Counter_1^{n+1}"; are these connected via the internal output feedback, or will I need to connect them directly? (I don't yet have a chip to test with, otherwise I could just experiment.)

**Comment by hneemann** [comment, marked as answer]
Building a top octave synthesizer is a nice idea. 
But if you want to do this with a GAL22v10, you will need a number of chips and you will have to divide the logic among the chips. Also, a GAL22v10 is not so good if you want to build logic with many FlipFlops, because each one has only ten. In addition, the programmers for the chips are comparatively expensive.
If you don't have a programmer yet, I would rather use a ready-made FPGA prototyping board, because it doesn't require a programmer, which makes it cheaper in sum.

If you use a GAL22v10, you do not have to make the feedback connections yourself. They already exist, and are activated when the chip is programmed.

**Comment by jimktrains** [comment]
Thanks!

For some reason I thought that they could be programmed with an EEPROM programmer like the TL866.

I'm just learning still, and definitely feeling the limitations when I go to build a JEDEC file :D It's fun none-the-less, and Digital is Just Nice to use to plan and simulate, even if I don't end up going down the CPLD route.

**Reply by hneemann** [reply]
If you already have an EEPROM programmer that can also program GALs, then you should definitely play around with a GAL22v10. 
As you already said: That's quite a lot of fun. :smile:

**Comment by xotmatrix** [comment]
TL866CS can program Lattice GAL22V10 PLDs. 
TL866-II Plus can also program Atmel ATF22V10 PLDs.
